@here/olp-sdk-core/test/unit/MultiPartUploadWrapper.test.ts

```diff
@@ -17,32 +17,655 @@
+
+            const completeMultipartUploadCalls = MockedBlobV2UploadRequest.completeMultipartUpload.getCalls();
+            expect(completeMultipartUploadCalls.length).eqls(1);
+
+            const completeMultipartUploadCallParams =
+                completeMultipartUploadCalls[0].args[0];
+
+            expect(completeMultipartUploadCallParams.parts).eqls([
+                { id: "mocked-uploaded-part-id", number: 1 },
+                { id: "mocked-uploaded-part-id", number: 2 }
+            ]);
+            expect(completeMultipartUploadCallParams.layerId).eqls(
+                "mocked-layer-id"
+            );
+            expect(completeMultipartUploadCallParams.multipartToken).eqls(
+                "mocked-multipartToken"
+            );
+            expect(completeMultipartUploadCallParams.billingTag).eqls(
+                undefined
+            );
+            expect(completeMultipartUploadCallParams.url).eqls(undefined);
+
+            BlobV2UploadRequestStub.restore();
         });
+    });
+
+    describe("Different inputs test", () => {
+        it("Upload to Blob", async () => {
+            // ======= Prepare ====== //
+
+            const NodeFileDataStub = sinon
+                .stub(NodeFileData, "fromPath")
+                .resolves(new BufferData(Buffer.from("test-data", "utf-8")));
+
+            const MockedBlobV2UploadRequest = {
+                startMultipartUpload: sinon.stub().resolves({
+                    multipartToken: "mocked-multipartToken"
+                }),
+                uploadPart: sinon
+                    .stub()
+                    .resolves({ id: "mocked-uploaded-part-id" }),
+                completeMultipartUpload: sinon.stub()
+            };
+
+            const BlobV2UploadRequestStub = sinon
+                .stub(BlobV2UploadRequest, "BlobV2UploadRequest")
+                .returns(MockedBlobV2UploadRequest);
+
+            // ======= Initialize ====== //
+            wrapper = new core.MultiPartUploadWrapper(
+                {
+                    blobVersion: "v2",
+                    catalogHrn: "mocked-catalog",
+                    contentType: "text/plain",
+                    handle: "mocked-key",
+                    layerId: "mocked-layer-id",
+                    contentEncoding: "gzip"
+                },
+                OlpClientSettingsStub
+            );
+
+            // ======== Call method. ======== //
+
+            const status = await wrapper.upload("fake path to file");
 
-        wrapper.upload({
-            blobVersion: 1,
-            catalogHrn: "mocked-catalog",
-            contentType: "text/plain",
-            handle: "mmocked-datahandle",
-            layerId: "mocked-layer-id"
+            expect(status).eqls(204);
+
+            NodeFileDataStub.restore();
+            BlobV2UploadRequestStub.restore();
         });
+    });
+
+    describe("Unsuccess cases", () => {

```

Comment:
"Unsuccess" - I don't think this is quite accurate. Maybe you mean "Unsuccessful cases".